simulator/virtual_machine.go

```diff
@@ -350,33 +343,43 @@ func (vm *VirtualMachine) createFile(spec string, name string, register bool) (*
 	return f, nil
 }
 
+// Rather than keep an fd open for each VM, open/close the log for each messages.
+// This is ok for now as we do not do any heavy VM logging.
+func (vm *VirtualMachine) logPrintf(format string, v ...interface{}) {
+	f, err := os.OpenFile(vm.log, os.O_APPEND|os.O_WRONLY, 0)
```

Comment:
do we need `O_CREAT` here?

Reply:
The file is created when the VM is, but yes we should, I've added the flag.